virtcontainers/sandbox.go

```diff
@@ -728,14 +728,14 @@ func fetchSandbox(ctx context.Context, sandboxID string) (sandbox *Sandbox, err
 
 	var config SandboxConfig
 
-	// Try to load sandbox config from old store at first.
-	c, ctx, err := loadSandboxConfigFromOldStore(ctx, sandboxID)
+	// Try to load sandbox config from new store at first.
```

Comment:
@keloyang, what was the reason of inverting the logic here?
I think we still should try to load from the old store at first.

Reply:
The root cause of the residue is loadSandboxConfigFromOldStore is called but old store file is not exist,  and if load sandbox from new store successfull, there is no need to call loadSandboxConfigFromOldStore.

Reply:
@amshinde, @egernst,

Let me ask some more experienced people for some help.

I think we still should try to load the old store first.

In that very same code path we call `createSandbox()`, which calls `newSandbox()` and there depending on whether we use the old store or not we'd call `store.NewCVSandboxStore()`.

My question here is, why we only call that for one case?

Reply:
@jodh-intel, maybe you can help me with this one? ^^

Reply:
@amshinde, @jodh-intel, @egernst,
ping about this question.

Reply:
@fidencio This patch makes sense to me. We have kept the old store code around for backward compatibility. But by checking for the old store and trying to initialize one and checking for failure while using the old store, we were inadvertently creating the directory structure under `/var/lib/vc/sbs/{sandbox_id}` and leaving that around after the sandbox is deleted. 
It makes sense to me to check for the new store first instead. I tested this patch and works as expected.